### PR TITLE
Add Beersheba documentation

### PR DESCRIPTION
**Pull request by halmamol**
This PR includes Beersheba documentation in the readthedocs. 

It can be checked here: https://pr008-sw-docs.readthedocs.io/en/latest/beersheba.html

### CODE REVIEW

File: docs/source/beersheba.rst [hunk 1]
```diff
@@ -1,5 +1,310 @@
...
+     - ``str`` (**None**, '**linear**', '**cubic**')
+     - Sensor interpolation method. If **None**, no interpolation will be applied. '**cubic**' not supported for 3D deconvolution.
+
+   * - ``psf_fname``
+     - ``string`` (filepath)
+     - File of the Point-Spread Function.
+
+   * - ``e_cut``
+     - ``float``
+     - Cut in relative value to the max voxel over the deconvolution output (order 1e-3).
```
Comment (martinperezmaneiro) [inline docs/source/beersheba.rst]: As the ``cut_type`` variable specifies the purpose of this number, its definition could be something like: ‘Cut to the voxel values of the deconvolution output. Then, cut_type specifies if the value is absolute or relative.’

File: docs/source/beersheba.rst [hunk 2]
```diff
@@ -1,5 +1,310 @@
...
+     - ``float``
+     - Stopping threshold (difference between iterations).
+
+   * - ``n_iterations``
+     - ``int``
+     - Number of Lucy-Richardson iterations to be applied if the ``iteration_tol`` criteria is not fulfilled before.
+
+   * - ``energy_type``
+     - ``str`` ('**E**', '**Ec**')
+     - Marks which energy type (see :doc:`esmeralda`) should be assigned to the deconvolved track.
```
Comment (martinperezmaneiro) [inline docs/source/beersheba.rst]: Maybe add: '('E' for uncorrected energy, 'Ec' for corrected energy; see Esmeralda)'

File: docs/source/beersheba.rst [hunk 3]
```diff
@@ -1,5 +1,310 @@
...
+     - Number of Lucy-Richardson iterations for gaussian in '**separate**' mode.
+
+
+.. _Beersheba workflow:
+
+Workflow
+--------
+
+Beersheba algorithm uses corrected energy hits from :doc:`esmeralda`. More concretely,
+the set of hits with a low threshold applied (``CHITS/highTh``). This is done in order
```
Comment (martinperezmaneiro) [inline docs/source/beersheba.rst]: I think this should be (``CHITS/lowTh``)

File: docs/source/beersheba.rst [hunk 4]
```diff
@@ -1,5 +1,310 @@
...
+Drop Isolated Sensors
+::::::::::::::::::::::::
+
+This cut drops single sensors if there are no adjacent non-zero neighbours in the same slice (in XY directions) that have seen
+light in the event. The distance of this cut can be configurable under the ``drop_dist`` parameter.
+This cut is applied to avoid filling the region between the main track and isolated SiPMs which fluctuate
+above the charge threshold ``q_cut`` by non-physical data in the subsequent interpolation step. After removing
+isolated sensors, the energy is distributed through the reminding hits so the total energy of the event remains the same.
+
+In the picture below [#]_ it is represented inpact of the :ref:`Cut Sensors <Cut-Sensors>` and :ref:`Drop isolated sensors <Drop-Sensors>`
```
Comment (martinperezmaneiro) [inline docs/source/beersheba.rst]: Typo: impact

File: docs/source/beersheba.rst [hunk 5]
```diff
@@ -1,5 +1,310 @@
...
+
+.. _Deconvolution:
+
+RL Deconvolution
+::::::::::::::::::
+
+Deconvolution is applied iteratively, maintaining the overall charge of the event. Currently, Beersheba
+runs a 2D RL deconvolution, this means that it is applied slice by slice. To do so, the code uses the
+corresponding z dependent PSF needed for each of the event slices, and maintains the overall charge
+of each slice constant. Depending how the PSF information is handled, two different *modes* of deconvolution
```
Comment (martinperezmaneiro) [inline docs/source/beersheba.rst]: Typo: Depending on how

File: docs/source/beersheba.rst (hunk 1 above)
Comment (halmamol) [reply docs/source/beersheba.rst]: I'll edit it!

File: docs/source/beersheba.rst (hunk 2 above)
Comment (halmamol) [reply docs/source/beersheba.rst]: Ok, I'll clarify it!

File: docs/source/beersheba.rst (hunk 3 above)
Comment (halmamol) [reply docs/source/beersheba.rst]: Good catch!

**Review by martinperezmaneiro — APPROVED** [review]
👍

**Pull request merged**